# Lidando com os valores NaN e Pré-processamento

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns

In [52]:
portfolio3 = pd.read_csv('dados/estaticos_portfolio3.csv')
market = pd.read_csv('dados/market_clean_analise_exploratoria.csv')
portfolio1 = pd.read_csv('dados/estaticos_portfolio1.csv')
portfolio2 = pd.read_csv('dados/estaticos_portfolio2.csv')

In [4]:
portfolio1 = portfolio1.drop(['Unnamed: 0'], axis = 1)
portfolio2 = portfolio2.drop(['Unnamed: 0'], axis = 1)
portfolio3 = portfolio3.drop(['Unnamed: 0'], axis = 1)
market = market.drop(['Unnamed: 0'], axis = 1)

In [5]:
print('Shape do Portfolio1 é', portfolio1.shape)
print('Shape do Portfolio2 é', portfolio2.shape)
print('Shape do Portfolio3 é', portfolio3.shape)
print('Shape do Market é', market.shape)

Shape do Portfolio1 é (555, 181)
Shape do Portfolio2 é (566, 1)
Shape do Portfolio3 é (265, 1)
Shape do Market é (462298, 40)


In [6]:
market.head()

id  fl_matriz  \
0  a6984c3ae395090e3bee8ad63c3758b110de096d5d8195...       True   
1  6178f41ade1365e44bc2c46654c2c8c0eaae27dcb476c4...       True   
2  4a7e5069a397f12fdd7fd57111d6dc5d3ba558958efc02...       True   
3  3348900fe63216a439d2e5238c79ddd46ede454df7b9d8...       True   
4  1f9bcabc9d3173c1fe769899e4fac14b053037b953a1e4...       True   

            de_natureza_juridica sg_uf natureza_juridica_macro  \
0  SOCIEDADE EMPRESARIA LIMITADA    RN  ENTIDADES EMPRESARIAIS   
1          EMPRESARIO INDIVIDUAL    PI                  OUTROS   
2          EMPRESARIO INDIVIDUAL    AM                  OUTROS   
3          EMPRESARIO INDIVIDUAL    AM                  OUTROS   
4          EMPRESARIO INDIVIDUAL    RN                  OUTROS   

                                          de_ramo             setor  \
0                         INDUSTRIA DA CONSTRUCAO  CONSTRUÇÃO CIVIL   
1              SERVICOS DE ALOJAMENTO/ALIMENTACAO           SERVIÇO   
2               TRANSPORTE, ARMAZENAGEM E CORREIO           SERVIÇO   
3                               SERVICOS DIVERSOS           SERVIÇO   
4  SERVICOS PROFISSIONAIS, TECNICOS E CIENTIFICOS           SERVIÇO   

   idade_empresa_anos idade_emp_cat  fl_me  ...  nm_micro_regiao  \
0           14.457534       10 a 15  False  ...            NATAL   
1            1.463014         1 a 5  False  ...         TERESINA   
2            7.093151        5 a 10  False  ...           MANAUS   
3            6.512329        5 a 10  False  ...           MANAUS   
4            3.200000         1 a 5  False  ...            NATAL   

   fl_passivel_iss  qt_socios idade_media_socios  qt_socios_feminino  \
0             True        2.0               44.0                 NaN   
1             True        1.0               27.0                 NaN   
2             True        1.0               32.0                 NaN   
3             True        1.0               36.0                 NaN   
4             True        1.0                NaN                 NaN   

          de_faixa_faturamento_estimado   de_faixa_faturamento_estimado_grupo  \
0  DE R$ 1.500.000,01 A R$ 4.800.000,00  DE R$ 1.500.000,01 A R$ 4.800.000,00   
1       DE R$ 81.000,01 A R$ 360.000,00       DE R$ 81.000,01 A R$ 360.000,00   
2                      ATE R$ 81.000,00                      ATE R$ 81.000,00   
3       DE R$ 81.000,01 A R$ 360.000,00       DE R$ 81.000,01 A R$ 360.000,00   
4       DE R$ 81.000,01 A R$ 360.000,00       DE R$ 81.000,01 A R$ 360.000,00   

   vl_faturamento_estimado_aux vl_faturamento_estimado_grupo_aux qt_filiais  
0                    3132172.8                         3132172.8          0  
1                     210000.0                          210000.0          0  
2                      50000.0                           50000.0          0  
3                     210000.0                          210000.0          0  
4                     210000.0                          210000.0          0  

[5 rows x 40 columns]

In [7]:
pd.set_option('display.max_columns', 500)

## Verificando a presença de outliers incoerente 

In [8]:
market.describe()

idade_empresa_anos  vl_total_veiculos_pesados_grupo  \
count       462298.000000                    460371.000000   
mean             9.948677                         3.591801   
std              9.615664                        72.600352   
min              0.016438                         0.000000   
25%              2.756164                         0.000000   
50%              6.704110                         0.000000   
75%             14.465753                         0.000000   
max            106.432877                      9782.000000   

       vl_total_veiculos_leves_grupo  nu_meses_rescencia  \
count                  460371.000000       417022.000000   
mean                       48.699177           25.007247   
std                      1206.696804            9.679799   
min                         0.000000            0.000000   
25%                         0.000000           22.000000   
50%                         0.000000           23.000000   
75%                         0.000000           25.000000   
max                    122090.000000           66.000000   

       empsetorcensitariofaixarendapopulacao      qt_socios  \
count                          318469.000000  347207.000000   
mean                             1308.005725       1.496326   
std                              1161.889222       3.276626   
min                               100.000000       1.000000   
25%                               673.230000       1.000000   
50%                               946.680000       1.000000   
75%                              1518.080000       2.000000   
max                             75093.840000     246.000000   

       idade_media_socios  qt_socios_feminino  vl_faturamento_estimado_aux  \
count       310696.000000       145029.000000                 4.347850e+05   
mean            42.816452            1.107117                 8.020911e+05   
std             12.626447            0.840194                 3.099979e+07   
min             -2.000000            1.000000                 0.000000e+00   
25%             33.000000            1.000000                 1.648512e+05   
50%             42.000000            1.000000                 2.100000e+05   
75%             51.000000            1.000000                 2.100000e+05   
max            127.000000          140.000000                 1.454662e+10   

       vl_faturamento_estimado_grupo_aux     qt_filiais  
count                       4.347850e+05  462298.000000  
mean                        3.367205e+08      28.737044  
std                         7.114614e+09     468.626094  
min                         4.104703e+04       0.000000  
25%                         1.854576e+05       0.000000  
50%                         2.100000e+05       0.000000  
75%                         2.100000e+05       0.000000  
max                         2.227618e+11    9647.000000

## Como podemos ver, a idade média dos sócios máxima é 127 anos e mínima -2 anos. Vamos tratar esse outliers.

In [9]:
market_media_ajuste = market.copy()

## Trocando os valores indevidos da idade

Através da base de dados do mercado, viemos a conclusão que seria coerente trocar as notas abaixo de 18 para 18 anos e acima de 70 para 70 anos.
Os valores faltantes, vamos preencher com a média da idade.

In [10]:
market_media_ajuste['idade_media_socios'].fillna(0, inplace=True)

In [11]:
def troca_media_idade(dado):
    if dado==0:
        return 42.816452
    elif dado < 18:
        return 18
    elif dado > 70:
        return 70
    else:
        return dado

In [12]:
market_media_ajuste['idade_media_socios'] = market_media_ajuste['idade_media_socios'].apply(troca_media_idade)

In [13]:
market_media_ajuste.shape

(462298, 40)

## Pré-Processamento

In [15]:
market_clean = market_media_ajuste.copy()

In [16]:
market_clean.dtypes

id                                        object
fl_matriz                                   bool
de_natureza_juridica                      object
sg_uf                                     object
natureza_juridica_macro                   object
de_ramo                                   object
setor                                     object
idade_empresa_anos                       float64
idade_emp_cat                             object
fl_me                                       bool
fl_sa                                       bool
fl_mei                                      bool
fl_ltda                                     bool
dt_situacao                               object
fl_st_especial                              bool
fl_email                                    bool
fl_telefone                                 bool
fl_rm                                       bool
nm_divisao                                object
nm_segmento                               object
fl_optante_simples  

In [17]:
market_clean.isna().sum()

id                                            0
fl_matriz                                     0
de_natureza_juridica                          0
sg_uf                                         0
natureza_juridica_macro                       0
de_ramo                                       0
setor                                      1927
idade_empresa_anos                            0
idade_emp_cat                                 0
fl_me                                         0
fl_sa                                         0
fl_mei                                        0
fl_ltda                                       0
dt_situacao                                   6
fl_st_especial                                0
fl_email                                      0
fl_telefone                                   0
fl_rm                                         0
nm_divisao                                 1927
nm_segmento                                1927
fl_optante_simples                      

## Preenchendo os valores numéricos NaN com a média

In [18]:
# Preenchendo os valores NaN's numericos
for i in market_clean.select_dtypes(['int64','float64']).columns:
    market_clean[i] = market_clean[i].fillna(market_clean[i].mean())

In [19]:
market_clean.isna().sum()

id                                           0
fl_matriz                                    0
de_natureza_juridica                         0
sg_uf                                        0
natureza_juridica_macro                      0
de_ramo                                      0
setor                                     1927
idade_empresa_anos                           0
idade_emp_cat                                0
fl_me                                        0
fl_sa                                        0
fl_mei                                       0
fl_ltda                                      0
dt_situacao                                  6
fl_st_especial                               0
fl_email                                     0
fl_telefone                                  0
fl_rm                                        0
nm_divisao                                1927
nm_segmento                               1927
fl_optante_simples                           0
vl_total_veic

## Preenchendo os valores bool e object NaN com False e 'other' respectivamente

In [20]:
fill_dict = market_clean.dtypes.to_dict()

In [21]:
for feature in fill_dict:
    if(fill_dict[feature] == bool):
        fill_dict[feature] = False
    elif(fill_dict[feature] == object):
        fill_dict[feature] = 'other'

In [22]:
market_clean.fillna(value=fill_dict, inplace=True)

In [23]:
market_clean.isna().sum()

id                                       0
fl_matriz                                0
de_natureza_juridica                     0
sg_uf                                    0
natureza_juridica_macro                  0
de_ramo                                  0
setor                                    0
idade_empresa_anos                       0
idade_emp_cat                            0
fl_me                                    0
fl_sa                                    0
fl_mei                                   0
fl_ltda                                  0
dt_situacao                              0
fl_st_especial                           0
fl_email                                 0
fl_telefone                              0
fl_rm                                    0
nm_divisao                               0
nm_segmento                              0
fl_optante_simples                       0
vl_total_veiculos_pesados_grupo          0
vl_total_veiculos_leves_grupo            0
sg_uf_matri

In [24]:
market_clean.head()

id  fl_matriz  \
0  a6984c3ae395090e3bee8ad63c3758b110de096d5d8195...       True   
1  6178f41ade1365e44bc2c46654c2c8c0eaae27dcb476c4...       True   
2  4a7e5069a397f12fdd7fd57111d6dc5d3ba558958efc02...       True   
3  3348900fe63216a439d2e5238c79ddd46ede454df7b9d8...       True   
4  1f9bcabc9d3173c1fe769899e4fac14b053037b953a1e4...       True   

            de_natureza_juridica sg_uf natureza_juridica_macro  \
0  SOCIEDADE EMPRESARIA LIMITADA    RN  ENTIDADES EMPRESARIAIS   
1          EMPRESARIO INDIVIDUAL    PI                  OUTROS   
2          EMPRESARIO INDIVIDUAL    AM                  OUTROS   
3          EMPRESARIO INDIVIDUAL    AM                  OUTROS   
4          EMPRESARIO INDIVIDUAL    RN                  OUTROS   

                                          de_ramo             setor  \
0                         INDUSTRIA DA CONSTRUCAO  CONSTRUÇÃO CIVIL   
1              SERVICOS DE ALOJAMENTO/ALIMENTACAO           SERVIÇO   
2               TRANSPORTE, ARMAZENAGEM E CORREIO           SERVIÇO   
3                               SERVICOS DIVERSOS           SERVIÇO   
4  SERVICOS PROFISSIONAIS, TECNICOS E CIENTIFICOS           SERVIÇO   

   idade_empresa_anos idade_emp_cat  fl_me  fl_sa  fl_mei  fl_ltda  \
0           14.457534       10 a 15  False  False   False    False   
1            1.463014         1 a 5  False  False    True    False   
2            7.093151        5 a 10  False  False    True    False   
3            6.512329        5 a 10  False  False   False    False   
4            3.200000         1 a 5  False  False   False    False   

  dt_situacao  fl_st_especial  fl_email  fl_telefone  fl_rm  \
0  2005-03-25           False      True         True   True   
1  2017-05-12           False      True         True   True   
2  2011-09-26           False     False         True   True   
3  2012-04-25           False      True         True   True   
4  2015-08-17           False      True         True   True   

                                          nm_divisao  \
0                            CONSTRUCAO DE EDIFICIOS   
1                                        ALIMENTACAO   
2                               TRANSPORTE TERRESTRE   
3  REPARACAO E MANUTENCAO DE EQUIPAMENTOS DE INFO...   
4  SERVICOS DE ARQUITETURA E ENGENHARIA TESTES E ...   

                                       nm_segmento  fl_optante_simples  \
0                                       CONSTRUCAO                True   
1                         ALOJAMENTO E ALIMENTACAO                True   
2                 TRANSPORTE ARMAZENAGEM E CORREIO                True   
3                    OUTRAS ATIVIDADES DE SERVICOS                True   
4  ATIVIDADES PROFISSIONAIS CIENTIFICAS E TECNICAS                True   

   vl_total_veiculos_pesados_grupo  vl_total_veiculos_leves_grupo  \
0                              0.0                            0.0   
1                              0.0                            0.0   
2                              0.0                            0.0   
3                              0.0                            0.0   
4                              0.0                            0.0   

  sg_uf_matriz de_saude_tributaria de_saude_rescencia  nu_meses_rescencia  \
0           RN               VERDE     ACIMA DE 1 ANO                23.0   
1           PI               CINZA          ATE 1 ANO                 9.0   
2           AM             AMARELO     ACIMA DE 1 ANO                26.0   
3           AM             AMARELO     ACIMA DE 1 ANO                26.0   
4           RN               VERDE     ACIMA DE 1 ANO                27.0   

  de_nivel_atividade  empsetorcensitariofaixarendapopulacao  \
0               ALTA                            1308.005725   
1              BAIXA                             468.930000   
2              MEDIA                             713.500000   
3              MEDIA                             475.730000   
4               ALTA                            

## Padronizando

In [25]:
from sklearn.preprocessing import StandardScaler

In [26]:
market_padronizacao = market_clean.copy()

In [27]:
market_padronizacao = market_padronizacao.reset_index()

In [28]:
market_padronizacao.drop('index',axis=1,inplace=True)

In [31]:
# Padronizando os dados
market_stdscaler = StandardScaler().fit_transform(market_padronizacao.select_dtypes(['float64','int64']))
market_stdscaler = pd.DataFrame(market_stdscaler)
colunas_int_float = market_padronizacao.select_dtypes(['float64','int64']).columns

market_stdscaler.columns = colunas_int_float

market_stdscaler = pd.concat([market_stdscaler, market_padronizacao.select_dtypes(['bool','object'])], axis=1)

In [32]:
market_stdscaler.shape

(462298, 40)

In [33]:
market_stdscaler.head()

idade_empresa_anos  vl_total_veiculos_pesados_grupo  \
0            0.468908                        -0.049577   
1           -0.882484                        -0.049577   
2           -0.296966                        -0.049577   
3           -0.357370                        -0.049577   
4           -0.701843                        -0.049577   

   vl_total_veiculos_leves_grupo  nu_meses_rescencia  \
0                      -0.040442           -0.218332   
1                      -0.040442           -1.741135   
2                      -0.040442            0.107983   
3                      -0.040442            0.107983   
4                      -0.040442            0.216755   

   empsetorcensitariofaixarendapopulacao  qt_socios  idade_media_socios  \
0                           3.206583e-14   0.177374            0.128755   
1                          -8.700907e-01  -0.174786           -1.571173   
2                          -6.164806e-01  -0.174786           -1.071194   
3                          -8.630394e-01  -0.174786           -0.671211   
4                           1.623920e+00  -0.174786            0.010406   

   qt_socios_feminino  vl_faturamento_estimado_aux  \
0                 0.0                     0.077506   
1                 0.0                    -0.019695   
2                 0.0                    -0.025017   
3                 0.0                    -0.019695   
4                 0.0                    -0.019695   

   vl_faturamento_estimado_grupo_aux  qt_filiais  \
0                          -0.048349   -0.061322   
1                          -0.048772   -0.061322   
2                          -0.048795   -0.061322   
3                          -0.048772   -0.061322   
4                          -0.048772   -0.061322   

                                                  id  fl_matriz  \
0  a6984c3ae395090e3bee8ad63c3758b110de096d5d8195...       True   
1  6178f41ade1365e44bc2c46654c2c8c0eaae27dcb476c4...       True   
2  4a7e5069a397f12fdd7fd57111d6dc5d3ba558958efc02...       True   
3  3348900fe63216a439d2e5238c79ddd46ede454df7b9d8...       True   
4  1f9bcabc9d3173c1fe769899e4fac14b053037b953a1e4...       True   

            de_natureza_juridica sg_uf natureza_juridica_macro  \
0  SOCIEDADE EMPRESARIA LIMITADA    RN  ENTIDADES EMPRESARIAIS   
1          EMPRESARIO INDIVIDUAL    PI                  OUTROS   
2          EMPRESARIO INDIVIDUAL    AM                  OUTROS   
3          EMPRESARIO INDIVIDUAL    AM                  OUTROS   
4          EMPRESARIO INDIVIDUAL    RN                  OUTROS   

                                          de_ramo             setor  \
0                         INDUSTRIA DA CONSTRUCAO  CONSTRUÇÃO CIVIL   
1              SERVICOS DE ALOJAMENTO/ALIMENTACAO           SERVIÇO   
2               TRANSPORTE, ARMAZENAGEM E CORREIO           SERVIÇO   
3                               SERVICOS DIVERSOS           SERVIÇO   
4  SERVICOS PROFISSIONAIS, TECNICOS E CIENTIFICOS           SERVIÇO   

  idade_emp_cat  fl_me  fl_sa  fl_mei  fl_ltda dt_situacao  fl_st_especial  \
0       10 a 15  False  False   False    False  2005-03-25           False   
1         1 a 5  False  False    True    False  2017-05-12           False   
2        5 a 10  False  False    True    False  2011-09-26           False   
3        5 a 10  False  False   False    False  2012-04-25           False   
4         1 a 5  False  False   False    False  2015-08-17           False   

   fl_email  fl_telefone  fl_rm  \
0      True         True   True   
1      True         True   True   
2     False         True   True   
3      True         True   True   
4      True         True   True   

                                          nm_divisao  \
0                            CONSTRUCAO DE EDIFICIOS   
1                                        ALIMENTACAO   
2                               TRANSPORTE TERRESTRE   
3  REPARACAO E MANUTENCAO DE EQUIPAMENTOS DE INFO...   
4  SERVICOS DE ARQUITETURA E ENGENHARIA T

In [34]:
market_clean_noId = market_stdscaler.drop('id',axis=1)

## LabelEnconder

In [35]:
categoricas = market_clean_noId.columns[market_clean_noId.dtypes == object]
nao_categoricas = market_clean_noId.columns[market_clean_noId.dtypes != object]
from sklearn import preprocessing
le = preprocessing.LabelEncoder()
market_clean_le = pd.DataFrame()
market_clean_le[categoricas] = market_clean_noId[categoricas].apply(lambda col: le.fit_transform(col))
market_clean_enconded = pd.concat([market_clean_noId[nao_categoricas],market_clean_le], axis =1)

In [36]:
market_clean_enconded.dtypes

idade_empresa_anos                       float64
vl_total_veiculos_pesados_grupo          float64
vl_total_veiculos_leves_grupo            float64
nu_meses_rescencia                       float64
empsetorcensitariofaixarendapopulacao    float64
qt_socios                                float64
idade_media_socios                       float64
qt_socios_feminino                       float64
vl_faturamento_estimado_aux              float64
vl_faturamento_estimado_grupo_aux        float64
qt_filiais                               float64
fl_matriz                                   bool
fl_me                                       bool
fl_sa                                       bool
fl_mei                                      bool
fl_ltda                                     bool
fl_st_especial                              bool
fl_email                                    bool
fl_telefone                                 bool
fl_rm                                       bool
fl_optante_simples  

In [37]:
# Convertendo as colunas bool em valores 1 e 0    
for i in market_clean_enconded.select_dtypes(bool).columns:
    market_clean_enconded[i] = np.where(market_clean_enconded[i] == True, 1,0)

In [38]:
market_clean_enconded.shape

(462298, 39)

In [39]:
market_clean_enconded.describe()

idade_empresa_anos  vl_total_veiculos_pesados_grupo  \
count        4.622980e+05                     4.622980e+05   
mean         3.297566e-16                     3.737208e-15   
std          1.000001e+00                     1.000001e+00   
min         -1.032924e+00                    -4.957709e-02   
25%         -7.480003e-01                    -4.957709e-02   
50%         -3.374255e-01                    -4.957709e-02   
75%          4.697628e-01                    -4.957709e-02   
max          1.003408e+01                     1.349699e+02   

       vl_total_veiculos_leves_grupo  nu_meses_rescencia  \
count                   4.622980e+05        4.622980e+05   
mean                   -1.080597e-16        1.206164e-15   
std                     1.000001e+00        1.000001e+00   
min                    -4.044185e-02       -2.720079e+00   
25%                    -4.044185e-02       -3.271032e-01   
50%                    -4.044185e-02       -2.183315e-01   
75%                    -4.044185e-02        3.864345e-16   
max                     1.013482e+02        4.458849e+00   

       empsetorcensitariofaixarendapopulacao     qt_socios  \
count                           4.622980e+05  4.622980e+05   
mean                           -3.793547e-15 -6.716026e-16   
std                             1.000001e+00  1.000001e+00   
min                            -1.252658e+00 -1.747865e-01   
25%                            -5.405334e-01 -1.747865e-01   
50%                             3.206583e-14 -1.747865e-01   
75%                             3.206583e-14  2.345859e-16   
max                             7.651320e+01  8.610450e+01   

       idade_media_socios  qt_socios_feminino  vl_faturamento_estimado_aux  \
count        4.622980e+05        4.622980e+05                 4.622980e+05   
mean        -1.321112e-15       -8.323592e-16                 2.080220e-15   
std          1.000001e+00        1.000001e+00                 1.000001e+00   
min         -2.471135e+00       -2.276209e-01                -2.668021e-02   
25%         -5.712151e-01       -2.276209e-01                -2.051127e-02   
50%          1.040564e-02        0.000000e+00                -1.969491e-02   
75%          3.287470e-01        0.000000e+00                -1.969491e-02   
max          2.728646e+00        2.951452e+02                 4.838423e+02   

       vl_faturamento_estimado_grupo_aux    qt_filiais      fl_matriz  \
count                       4.622980e+05  4.622980e+05  462298.000000   
mean                        7.817816e-15 -8.093637e-15       0.937127   
std                         1.000001e+00  1.000001e+00       0.242734   
min                        -4.879659e-02 -6.132197e-02       0.000000   
25%                        -4.877211e-02 -6.132197e-02       1.000000   
50%                        -4.877211e-02 -6.132197e-02       1.000000   
75%                        -4.876670e-02 -6.132197e-02       1.000000   
max                         3.223715e+01  2.052441e+01       1.000000   

               fl_me          fl_sa         fl_mei        fl_ltda  \
count  462298.000000  462298.000000  462298.000000  462298.000000   
mean        0.002628       0.018239       0.326413       0.002687   
std         0.051198       0.133816       0.468901       0.051763   
min         0.000000       0.000000       0.000000       0.000000   
25%         0.000000       0.000000       0.000000       0.000000   
50%         0.000000       0.000000       0.000000       0.000000   
75%         0.000000       0.000000       1.000000       0.000000   
max         1.000000       1.000000       1.000000       1.000000   

       fl_st_especial       fl_email    fl_telefone          fl_rm  \
count   462298.000000  462298.000000  462298.000000  462298.000000   
mean         0.000147       0.445751       0.725653       0.487822   
std          0.012127       0.497049       0.446185       0.499852   
min          0.000000       0.000000       0.000000       0.000000   
25%          0.

In [40]:
market_clean_enconded.head()

idade_empresa_anos  vl_total_veiculos_pesados_grupo  \
0            0.468908                        -0.049577   
1           -0.882484                        -0.049577   
2           -0.296966                        -0.049577   
3           -0.357370                        -0.049577   
4           -0.701843                        -0.049577   

   vl_total_veiculos_leves_grupo  nu_meses_rescencia  \
0                      -0.040442           -0.218332   
1                      -0.040442           -1.741135   
2                      -0.040442            0.107983   
3                      -0.040442            0.107983   
4                      -0.040442            0.216755   

   empsetorcensitariofaixarendapopulacao  qt_socios  idade_media_socios  \
0                           3.206583e-14   0.177374            0.128755   
1                          -8.700907e-01  -0.174786           -1.571173   
2                          -6.164806e-01  -0.174786           -1.071194   
3                          -8.630394e-01  -0.174786           -0.671211   
4                           1.623920e+00  -0.174786            0.010406   

   qt_socios_feminino  vl_faturamento_estimado_aux  \
0                 0.0                     0.077506   
1                 0.0                    -0.019695   
2                 0.0                    -0.025017   
3                 0.0                    -0.019695   
4                 0.0                    -0.019695   

   vl_faturamento_estimado_grupo_aux  qt_filiais  fl_matriz  fl_me  fl_sa  \
0                          -0.048349   -0.061322          1      0      0   
1                          -0.048772   -0.061322          1      0      0   
2                          -0.048795   -0.061322          1      0      0   
3                          -0.048772   -0.061322          1      0      0   
4                          -0.048772   -0.061322          1      0      0   

   fl_mei  fl_ltda  fl_st_especial  fl_email  fl_telefone  fl_rm  \
0       0        0               0         1            1      1   
1       1        0               0         1            1      1   
2       1        0               0         0            1      1   
3       0        0               0         1            1      1   
4       0        0               0         1            1      1   

   fl_optante_simples  fl_passivel_iss  de_natureza_juridica  sg_uf  \
0                   1                1                    60      4   
1                   1                1                    17      3   
2                   1                1                    17      1   
3                   1                1                    17      1   
4                   1                1                    17      4   

   natureza_juridica_macro  de_ramo  setor  idade_emp_cat  dt_situacao  \
0                        2       11      2              1         2709   
1                        5       21      4              0         6811   
2                        5       32      4              3         4783   
3                        5       25      4              3         4994   
4                        5       27      4              0         6183   

   nm_divisao  nm_segmento  sg_uf_matriz  de_saude_tributaria  \
0          32           10            19                    4   
1           3            3            16                    2   
2          86           20             2                    0   
3          74           17             2                    0   
4          77            8            19                    4   

   de_saude_rescencia  de_nivel_atividade  nm_meso_regiao  nm_micro_regiao  \
0                   0                   0               6               48   
1                   1                   1               4               69   
2                   0                   2               2               43   
3                   0                   2               2               43   
4               

In [41]:
market_clean_enconded['id'] = market_dummies_stdscaler['id']

In [43]:
market_clean_enconded.to_csv('market_clean_enconded.csv')

## Analisando se os ids estão presentes nos 3 portfólios, se está só em 2 ou só em 1 portfólio

# Merge nos portfólios  com o market

In [44]:
df_portfolio1 = pd.merge(portfolio1.id, market_clean_enconded, on='id')
df_portfolio2 = pd.merge(portfolio2, market_clean_enconded, on='id')
df_portfolio3 = pd.merge(portfolio3, market_clean_enconded, on='id')

In [45]:
df_portfolio1.to_csv('df_portfolio1.csv')
df_portfolio2.to_csv('df_portfolio2.csv')
df_portfolio3.to_csv('df_portfolio3.csv')

In [46]:
port1 = list(df_portfolio1['id'])
port2 = list(df_portfolio2['id'])
port3 = list(df_portfolio3['id'])

In [47]:
port1_port2 = [x for x in port1 if x in port2]
port1_port3 = [x for x in port1 if x in port3]
port2_port3 = [x for x in port2 if x in port3]
port1_port2_port3 = [x for x in port1 if x in port2 and port3]

In [48]:
print(f"Port1 and Port2: {len(port1_port2)}")
print(f"Port1 and Port3: {len(port1_port3)}")
print(f"Port2 and Port3: {len(port2_port3)}")
print(f"Port1, Port2 and Port3: {len(port1_port2_port3)}")

Port1 and Port2: 0
Port1 and Port3: 0
Port2 and Port3: 108
Port1, Port2 and Port3: 0
